tools/respecDocWriter.js

```diff
@@ -189,13 +189,13 @@ async function evaluateHTML() {
       // Document references an older version of ReSpec that does not yet
       // have the "core/exporter" module. Try with the old "ui/save-html"
       // module.
-      const { exportDocument } = await new Promise((resolve, reject) => {
-        require(["ui/save-html"], resolve, reject);
```

Comment:
There is some guarantee now that the modules do exist so this won't fail in practice but, unless the `require` here is not the one from RequireJS, I note it [does accept an error callback](http://requirejs.org/docs/api.html#errbacks).

Not rejecting the promise was precisely what made RespecDocWriter crash badly initially: the promise stayed in pending mode forever, and `evaluateHTML` simply never returned, so there was no way to tell programmatically that the generation had failed (except with a timeout). That behavior is easy to reproduce, just require a module here that does not exist!